### PR TITLE
feat(tenants): expose booking limits in the API

### PR DESCRIPTION
**Pull request by chillfox**
- return booking_limits in #index
- add #current_limits, #show_limits, #update_limits
- add `limit_override` query param to Bookings #create and #update

### CODE REVIEW

File: src/controllers/tenants.cr [hunk 1]
```diff
@@ -39,6 +39,22 @@ class Tenants < Application
     Tenant.find!(params["id"].to_i64).delete
   end
 
+  get "/current_limits", :current_limits do
```
Comment (chillfox) [inline src/controllers/tenants.cr:42]: @tassja how can I expose this route to all logged-in users? (https://github.com/PlaceOS/staff-api/issues/161)

Comment (tassja) [reply src/controllers/tenants.cr:42]: Is that route not exposed to all users? There isn't any auth or log in checks? @stakach

Comment (chillfox) [reply src/controllers/tenants.cr:42]: I would assume that the `before_action :admin_only` applies to all routes in the controller, so I guess my question is, how do I exempt the route from that.

Comment (tassja) [reply src/controllers/tenants.cr:42]: Oh sorry missed that

Comment (tassja) [reply src/controllers/tenants.cr:42]: `before_action :admin_only, except: :current_limits`

Comment (chillfox) [reply src/controllers/tenants.cr:42]: Thanks

File: src/controllers/tenants.cr [hunk 2]
```diff
@@ -1,7 +1,7 @@
 class Tenants < Application
   base "/api/staff/v1/tenants"
 
-  before_action :admin_only
+  before_action :admin_only, except: [:current_limits, :show_limits]
```
Comment (tassja) [inline src/controllers/tenants.cr:4]: I would add a test for this to ensure any user can access

Comment (stakach) [reply src/controllers/tenants.cr:4]: probably no need to test for this as it's being tested at the spider-gazelle level

File: src/controllers/bookings.cr [hunk 3]
```diff
@@ -81,8 +81,10 @@ class Bookings < Application
     booking.booked_by_name = user.name
 
     # check concurrent bookings don't exceed booking limits
-    booking_limits = check_booking_limits(tenant, booking)
-    render :conflict, json: booking_limits if booking_limits
+    unless query_params["limit_override"]? == "true"
```
Comment (stakach) [inline src/controllers/bookings.cr:84]: As per the comment https://github.com/place-technology/suncorp-ntt/issues/175#issuecomment-1068824084
we want to allow a per-user custom limit using this param

**Review by stakach — CHANGES_REQUESTED** [review]

**Review by stakach — APPROVED** [review]
LGTM

**Pull request merged**